File: 2406-divide-intervals-into-minimum-number-of-groups/2406-divide-intervals-into-minimum-number-of-groups.java
class Solution {
    public int minGroups(int[][] intervals) {
        Arrays.sort(intervals, (a,b) -> a[0] - b[0]);
        final PriorityQueue<Integer> pq = new PriorityQueue<>();

        for (final int[] num : intervals) {
            if (!pq.isEmpty() && pq.peek() < num[0]) {
                pq.poll();
            }
            pq.offer(num[1]);
        }

        return pq.size();
    }
}

File: 2406-divide-intervals-into-minimum-number-of-groups/2406-divide-intervals-into-minimum-number-of-groups.py
class Solution:
    def minGroups(self, intervals: List[List[int]]) -> int:
        # intervals.sort(key=lambda x:x[-1])
        intervals.sort()
        res = []
        for st,ed in intervals:
            if not res or res[0][0] >= st:
                heapq.heappush(res, (ed, st))
            else:
                heapq.heapreplace(res, (ed, st))

        return len(res)
